### PR TITLE
feat: allow $.Deferred for new-cap

### PR DESCRIPTION
**Pull request by koba04**
I've noticed that `$.Deferred()` is reported as `new-cap` violation so I want to add `capIsNewExceptions` for `$.Deferred()`

### CODE REVIEW

File: lib/base.js [hunk 1]
```diff
@@ -213,7 +213,7 @@ module.exports = {
     "max-nested-callbacks": ["warn", 5],
     "max-params": ["warn", 7],
     "max-statements": ["warn", 40],
-    "new-cap": "warn",
+    "new-cap": ["warn", { capIsNewExceptions: ["$.Deferred"] }],
```
Comment (teppeis) [inline lib/base.js]: @koba04 `$.Deferred()` is ok, but `jQuery.Deferred()` is not. Both are allowd in globals.
https://github.com/cybozu/eslint-config/blob/05422fec18c159ef9f31c16a1ab24915491336bb/globals/kintone.js#L4-L5

Is this better?

```suggestion
    "new-cap": ["warn", { capIsNewExceptions: ["Deferred"] }],
```

Comment (koba04) [reply lib/base.js]: @teppeis 
Thank you for your suggestion! I've updated the option and a test.
Sorry, I've missed that the suggestion is used GitHub's suggestion feature 🙏

Comment (teppeis) [reply lib/base.js]: It's hard to notice :)

**Comment by teppeis** [comment]
LGTM!

**Comment by koba04** [comment]
Thanks!

**Pull request merged**